sdk/servicebus/service-bus/test/internal/tracing.spec.ts

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import chai from "chai";
+import { getTracer, NoOpSpan, TestSpan, TestTracer } from "@azure/core-tracing";
+import { CanonicalCode, SpanOptions } from "@opentelemetry/api";
+import {
+  ServiceBusMessage,
+  ServiceBusMessageImpl,
+  ServiceBusReceivedMessage
+} from "../../src/serviceBusMessage";
+import {
+  createAndEndProcessingSpan,
+  createProcessingSpan,
+  instrumentServiceBusMessage,
+  TRACEPARENT_PROPERTY
+} from "../../src/diagnostics/instrumentServiceBusMessage";
+import { OperationOptionsBase, trace } from "../../src/modelsToBeSharedWithEventHubs";
+import { setTracerForTest } from "../utils/misc";
+import { SpanKind } from "@opentelemetry/api";
+import { BatchingReceiverLite } from "../../src/core/batchingReceiver";
+import { Receiver } from "rhea-promise";
+import { createConnectionContextForTests } from "./unittestUtils";
+import sinon from "sinon";
+import { ServiceBusReceiverImpl } from "../../src/receivers/receiver";
+import { OnMessage } from "../../src/core/messageReceiver";
+import { ServiceBusSessionReceiverImpl } from "../../src/receivers/sessionReceiver";
+import { MessageSession } from "../../src/session/messageSession";
+const should = chai.should();
+const assert = chai.assert;
+
+describe("Tracing tests", () => {
```

Comment:
Can you also add changelog?

Reply:
Done. Looks like we've got all the major methods with tracing at this point so I just merged yours and my changelog entry.